gymnasium/core.py

```diff
@@ -16,23 +16,18 @@
 import numpy as np
 
 from gymnasium import spaces
-from gymnasium.logger import warn
 from gymnasium.utils import seeding
 
 if TYPE_CHECKING:
     from gymnasium.envs.registration import EnvSpec
 
-if sys.version_info[0:2] == (3, 6):
-    warn(
-        "Gymnasium minimally supports python 3.6 as the python foundation not longer supports the version, please update your version to 3.7+"
-    )
 
 ObsType = TypeVar("ObsType")
 ActType = TypeVar("ActType")
 RenderFrame = TypeVar("RenderFrame")
 
 
-class Env(Generic[ObsType, ActType]):
+class Env(Generic[ObsType, ActType], ABC):
```

Comment:
So this is something I've been contemplating for some time, whether Env should be an ABC, and I was always leaning more towards "No". It kind of is, and kind of isn't. Notice that, for example, `Env.reset` isn't in fact an abstract method because it does have some logic in it. In general, ABCs are meant to be empty skeletons, but we have a good chunk of built-in implementations. So I'm still kinda on the fence, and crucially I'm not 100% sure what runtime implications this has exactly.

Reply:
Yep, I'd keep it non-abstract altogether. I'd also strongly consider moving towards asking env developers to call `super().step(action)` and similarly for all the other methods, but this would be a relatively complex procedure that I don't have sufficient justification and motivation to push for at the moment.

Reply:
That makes sense and agree that `super().step(action)` could work but there isn't much good reasons for it other than we do it for `reset`